samtranslator/model/sam_resources.py

```diff
@@ -633,9 +633,10 @@ def _construct_lambda_layer(self, intrinsics_resolver):
         """
         retention_policy_value = self._get_retention_policy_value(intrinsics_resolver)
 
-        retention_policy = {
-            'DeletionPolicy': retention_policy_value
-        }
+        retention_policy = self.get_passthrough_resource_attributes()
```

Comment:
Could you rename `retention_policy` here since it could contain both `DeletionPolicy` and `Condition` attributes?